Marlin/src/module/ft_motion.cpp

```diff
@@ -223,16 +223,19 @@ void FTMotion::loop() {
 
   // Interpolation (generation of step commands from fixed time trajectory).
   while (batchRdyForInterp
-    && (stepperCmdBuffItems() < (FTM_STEPPERCMD_BUFF_SIZE) - (FTM_STEPS_PER_UNIT_TIME))) {
+    && (stepperCmdBuffItems() < (FTM_STEPPERCMD_BUFF_SIZE) - (FTM_STEPS_PER_UNIT_TIME))
+  ) {
     generateStepsFromTrajectory(interpIdx);
     if (++interpIdx == FTM_BATCH_SIZE) {
       batchRdyForInterp = false;
       interpIdx = 0;
     }
   }
 
-  // Report busy status to planner.
+  // Set busy status for use by planner.busy()
+  const bool oldBusy = busy;
   busy = (stepperCmdBuffHasData || blockProcRdy || batchRdy || batchRdyForInterp);
+  if (oldBusy && !busy) moving_axis_flags.reset();
```

Comment:
It's inexpensive to clear these flags, but without the check for a change of busy status they are cleared on every call to `idle()` where `busy` is false. With this change they are only cleared on a transition from busy to not-busy.